### PR TITLE
Fix to logic determining if context.matched in response

### PR DESCRIPTION
**Pull request by dchandan**
**Description:**
Currently, if an API does not return the `context.matched` key in the search response then `ItemSearch.matched` fails with a `KeyError`. This is not ideal, because if for any reason appropriate information needed to determine the number of matched results is not present in the response, then according to the function's description, the function should return `None`. This PR makes a minor fix to ameliorate the issue.

P.S. I realize now that I've misspelled "Matched" as "Marched" in my branch name. Sorry.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by dchandan** [comment]
Okay, I just noticed that my editor did some black formatting as well. I hope this is OK and there are no rule violations with the performed formatting.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+- Fix to logic determining if `context.matched` in `ItemSearch.matched` response
```
Comment (gadomski) [inline CHANGELOG.md:10]:
```suggestion
- Fix to logic determining if `context.matched` in `ItemSearch.matched` response [#646](https://github.com/stac-utils/pystac-client/pull/646)
```

**Comment by gadomski** [comment]
@dchandan yeah, the formatting broke CI. Please use **pre-commit** to ensure the hooks run correctly: https://pystac-client.readthedocs.io/en/stable/contributing.html#code-quality-checks.

Otherwise, looks good, thank you!

**Comment by dchandan** [comment]
> @dchandan yeah, the formatting broke CI. Please use **pre-commit** to ensure the hooks run correctly: https://pystac-client.readthedocs.io/en/stable/contributing.html#code-quality-checks.
> 
> Otherwise, looks good, thank you!

Thanks for the clarification @gadomski. pre-commit issue should be fixed now.

**Review by gadomski — APPROVED** [review]

**Pull request merged**